spec/unit/actions/synchronize_users_spec.rb

```diff
@@ -72,5 +76,39 @@
         subject.class.call(params)
       end
     end
+
+    context 'when using mode sync_domain_client_workspaces' do
+      let(:segment) { double('segment') }
+      let(:segment_uri) { 'segment_uri' }
+      let(:params) do
+        params = {
+          GDC_GD_CLIENT: client,
+          input_source: 'foo',
+          domain: 'bar',
+          gdc_logger: logger,
+          sync_mode: 'sync_domain_client_workspaces',
+          segments_filter: [segment]
+        }
+        GoodData::LCM2.convert_to_smart_hash(params)
+      end
+      let(:organization_not_in_segment) { double('organization') }
+
+      before do
+        allow(File).to receive(:open).and_return("client_id\n123456789")
+        allow(domain).to receive(:clients).with(:all, nil).and_return([organization, organization_not_in_segment])
+        allow(domain).to receive(:clients).with('123456789', nil).and_return(organization)
+        allow(segment).to receive(:uri).and_return(segment_uri)
+        allow(organization).to receive(:segment_uri).and_return(segment_uri)
+        allow(organization_not_in_segment).to receive(:segment_uri).and_return('another-segment-uri')
+        allow(organization).to receive(:project).and_return(project)
+        allow(organization).to receive(:client_id).and_return('123456789')
+      end
+
+      it 'only uses clients from segments in segments filter' do
```

Comment:
Identical code found in synchronize_user_filters_spec.rb:73, synchronize_users_spec.rb:107 (mass = 72)